app/assets/javascripts/student_profile/page_container.jsx

```diff
@@ -152,6 +153,33 @@ function fromPair(key, value) {
         .fail(this.onSaveNotesFail);
     },
 
+    onClickSaveTransitionNote: function(noteParams) {
```

Comment:
This might feel too late, but something to consider here is cutting-pasting the changes in this component into `<TransitionNote />`.  The `TransitionNote` UI code is nicely wrapped up in its own component, and it could hold these callbacks and state itself, keeping it isolated and safer and cheaper to iterate over there.  Then you don't have to worry about any interactions with anything in this complicated bigger component and it's easier to test in isolation too.

Alternately you could cut-paste this code into a ` <TransitionNoteContainer />` that the main profile page UI calls, and that has these callbacks and state, and then it passes those all as props when rendering `<TransitionNote />` as it is.